File: 1/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"unicode"
	"strconv"
	"strings"
)

func part_one() {
	readFile, err := os.Open("input")
  
    if err != nil {
        fmt.Println(err)
    }
    fileScanner := bufio.NewScanner(readFile)
 
    fileScanner.Split(bufio.ScanLines)
	resultado := 0
    for fileScanner.Scan() {
		cleared := ""
		for _ , char := range fileScanner.Text() {
			if unicode.IsDigit(char) {
				cleared += string(char)
			}
		}
		partialres := string(string(cleared[0]) + string(cleared[len(cleared)-1:]))
		num, _ := strconv.Atoi(partialres)
		resultado += num

    }
	fmt.Printf(strconv.Itoa(resultado))

    readFile.Close()
}

func part_two() {
	readFile, err := os.Open("input")
  
    if err != nil {
        fmt.Println(err)
    }
    fileScanner := bufio.NewScanner(readFile)
 
    fileScanner.Split(bufio.ScanLines)
	resultado := 0
	listanums := [9]string{"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}

    for fileScanner.Scan() {
		cleared := ""
		for i , char := range fileScanner.Text() {
			if unicode.IsDigit(char) {
				cleared += string(char)
			} else {
				for index, element := range listanums {
					if strings.HasPrefix(fileScanner.Text()[i:], element) {
						cleared += strconv.Itoa(index+1)
					}
				}
			}
		}
		partialres := string(string(cleared[0]) + string(cleared[len(cleared)-1:]))
		num, _ := strconv.Atoi(partialres)
		resultado += num

    }
	fmt.Printf(strconv.Itoa(resultado))

    readFile.Close()
}

func main() {
 
    part_two()
}

File: 1/day1.py

def part_two():
    result = 0
    dic = {
                'one': "o1e", 
                'two': "t2o",
                'three': "t3e",
                'four': "f4r",
                'five': "f5e",
                'six': "s6x",
                'seven': "s7n",
                'eight': "e8t",
                'nine': "n9e"
                }
    
    with open("input") as f:
        for file in f.readlines():
                       
            for k,v in dic.items():
                file = file.replace(k,v)
            digits = ''.join(filter(str.isdigit, file))

            result += int(digits[0]+digits[-1])
            
    return result
    
def part_one():
    result = 0
    with open("input") as f:
        for file in f.readlines():
            digits = ''.join(filter(str.isdigit, file))
            result += int(digits[0]+digits[-1])
            
            
    return result

print(part_two())